vermouth/molecule.py

```diff
@@ -981,6 +1069,16 @@ class Link(Block):
     """
     Template link between two residues.
 
+    Two links are equal if:
+    
+    * the underlying molecules are equal
+    * the names are equal
+    * the negative edges ("non_edges") are the equal regardless of order
+    * the interactions to remove are the same and in the same order
+    * the meta variables are equals regardless of order
+    * the pattern definitions are equals and in the same order
+    * the features are equals regardless of order
```

Comment:
I'm going to need comments on *why* order does or does not matter, also for Molecules and Blocks